source/advanced-topics.adoc

```diff
@@ -4,6 +4,187 @@
 This chapter covers topics that are beyond the scope of the introductory
 tutorial but are often useful in real-world RPM packaging.
 
+[[Signing-Packages]]
+== Signing Packages
+
+Signing a package is a way to secure the package for an end user. Secure transport can be achieved with implementation of the HTTPS protocol, which can be done when the package was downloaded just before installing. However, the packages are often downloaded in advance and stored in local repositories before they are used. The packages are signed to make sure no third party can alter the content of a package.
+
+There are three ways to sign a package:
+
+* xref:Adding-a-Signature-to-a-Package[Adding a signature to an already existing package].
+
+* xref:Replacing-a-Package-Signature[Replacing the signature on an already existing package].
+
+* xref:Build-time-signing[Signing a package at build-time].
+
+[[Adding-a-Signature-to-a-Package]]
+=== Adding a Signature to a Package
+
+In most cases packages are build without a signature. The signature is added just before release of the package.
+
+In order to add another signature to the package use the --addsign option.
+Having more than one signature makes it possible to record the package's path of ownership
+from the package builder to the end-user.
+
+.Example:
+----
+As an example, a division of a company creates a package and signs it with the division's key. The company's headquarters then checks the package's signature and adds the corporate signature to the package, stating that the signed package is authentic.
+
+With two signatures, the package makes its way to a retailer. The retailer checks the signatures and, if they check out, adds their signature as well.
+
+The package now makes its way to a company that wishes to deploy the package. After checking every signature on the package, they know that it is an authentic copy, unchanged since it was first created. Depending on the deploying company's internal controls, they may choose to add their own signature, to reassure their employees that the package has received their corporate approval.
+----
+
+The output from the --addsign option:
+
+[source,bash]
+----
+# rpm --addsign blather-7.9-1.i386.rpm
```

Comment:
Should this be run as root (#) or as a normal user ($)? Same for other commands.

Reply:
This might be a question for @xsuchy or others...

Reply:
It should be run by the user who is adding the signature. So normal user.